iceoryx_hoofs/include/iceoryx_hoofs/internal/objectpool/objectpool.hpp

```diff
@@ -435,6 +436,17 @@ class ObjectPool
 
         return m_freeIndex;
     }
+
+    // private member accessors for unit tests
+    char* getFirstPtr() const
```

Comment:
Nitpick: In gernal in iceoryx we try to always `NameWhatsInsideWithoutAbbrevations`. Hence, here my personal preference would be `getFirstPointer`. But it's not a biggy as ptr is a common abbrevation.